framework/src/modules/chain/logic/init_transaction.js

```diff
@@ -163,41 +49,21 @@ class Transaction {
 
 	// TODO: remove after https://github.com/LiskHQ/lisk/issues/2424
 	dbRead(raw) {
-		if (!raw.t_id) {
+		if (
+			raw.t_id === undefined ||
+			raw.t_id === null ||
+			raw.t_type === undefined ||
+			raw.t_type === null
+		) {
 			return null;
 		}
+		const transactionClass = this.transactionClassMap.get(raw.t_type);
 
-		const transactionJSON = {
-			id: raw.t_id,
-			height: raw.b_height,
-			blockId: raw.b_id || raw.t_blockId,
-			type: parseInt(raw.t_type),
-			timestamp: parseInt(raw.t_timestamp),
-			senderPublicKey: raw.t_senderPublicKey,
-			requesterPublicKey: raw.t_requesterPublicKey,
-			senderId: raw.t_senderId,
-			recipientId: raw.t_recipientId,
-			recipientPublicKey: raw.m_recipientPublicKey || null,
-			amount: raw.t_amount,
-			fee: raw.t_fee,
-			signature: raw.t_signature,
-			signSignature: raw.t_signSignature,
-			signatures: raw.t_signatures ? raw.t_signatures.split(',') : [],
-			confirmations: parseInt(raw.confirmations || 0),
-			asset: {},
-		};
-
-		const assetDbRead = assetDbReadMap.get(transactionJSON.type);
-
-		if (!assetDbRead) {
-			throw new Error(`Unknown transaction type ${transactionJSON.type}`);
+		if (!transactionClass) {
+			return null;
 		}
 
-		const asset = assetDbRead(raw);
-
-		if (asset) {
-			transactionJSON.asset = Object.assign(transactionJSON.asset, asset);
-		}
+		const transactionJSON = transactionClass.prototype.dbRead(raw);
```

Comment:
prefer to instanciate and call dbRead, otherwise it should be static

Reply:
Since you first need the JSON to intialize, and you call `dbRead` to get it calls `assetDbRead`, I would have to first strip the `raw.asset` from the db object, then instantiate. I think this way is cleaner. Making it `static` does not work since `dbRead` calls `this.assetDbRead`.